tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -100,6 +101,24 @@ static void CreateOperationResultFromOutputs(
                                    ppResult);
 }
 
+struct DefaultFPEnvScope
+{
+  // _controlfp_s is non-standard and <cfenv>.feholdexceptions doesn't work on windows...?
+#ifdef _WIN32
+  unsigned int previousValue;
+  DefaultFPEnvScope() {
+    // No exceptions, preserve denormals & round to nearest.
+    _controlfp_s(&previousValue, _MCW_EM | _DN_SAVE | _RC_NEAR, _MCW_EM | _MCW_DN | _MCW_RC);
+  }
+  ~DefaultFPEnvScope() {
+    unsigned int newValue;
+    _controlfp_s(&newValue, previousValue, _MCW_EM | _MCW_DN | _MCW_RC);
```

Comment:
what happens if `_controlfp_s` returns an error code? same comment for other usage.

Reply:
The destructor  case is difficult because we might already be handling another error since this is a cleanup routine. We could ignore, assert or failfast, how do we usually handle this?

Reply:
We talked offline. someway to communicate failure would be good if possible.